timeseries/src/autogluon/timeseries/evaluator.py

```diff
@@ -195,11 +195,17 @@ def check_get_evaluation_metric(
             return TimeSeriesEvaluator.DEFAULT_METRIC
         return metric
 
-    def __call__(self, data: TimeSeriesDataFrame, predictions: TimeSeriesDataFrame) -> float:
+    def cache_historic_metrics(self, data_past: TimeSeriesDataFrame):
+        self.historic_naive_1_error = in_sample_naive_1_error(y_history=data_past[self.target_column])
```

Comment:
A couple of notes about caching.

- First, should this method and the instance variable be private?
- Strictly, this is not a cache. A cache is often something that a user doesn't explicitly interact with. How about just `save_past_metrics` ?

Reply:
- Agree, accepted all renaming suggestions & made `_past_naive_1_error` a private attribute.
- We call `cache_historic_metrics ` (now renamed to `save_past_metrics`) inside `TimeSeriesEnsembleSelection._fit`, so this one should be public.